README.md

```diff
@@ -1,97 +1,161 @@
+[![npm version](https://img.shields.io/npm/v/xxx.svg?style=flat)](https://www.npmjs.com/package/xxx)
+-->
+[![build status](https://github.com/maxGraph/maxGraph/workflows/Build/badge.svg)](https://github.com/maxGraph/maxGraph/actions/workflows/build.yml)
 
-### Setting up local development environment
+`maxGraph` is a client side TypeScript diagramming library that uses SVG and HTML for rendering.
 
-In the project root directory, execute
+The `maxGraph` library uses no third-party software, it requires no plugins and can be integrated in virtually any framework (it's vanilla JS).
 
-```sh
-$ npm install
-```
+**Browser support**: Chrome, Edge, Firefox, Safari, Chromium based browsers (Brave, Opera, ....)
+on **mobile** and **desktop**.
 
-To watch the core package, execute
+`maxGraph` is the successor of [mxGraph](https://github.com/jgraph/mxgraph) which is now end of life.
+At first, it provides the same features as mxGraph and adds
+- TypeScript support
+- maintained npm package
+- modern modular, tree shakable, version of mxGraph to reduce the whole package size
 
-```sh
-$ npm run dev
-```
+New features will follow.
 
-and select `@mxgraph/core`.
 
-To run the html(vanilla-js) version of storybook, execute
+## Project status
 
-```sh
-$ npm run dev
+maxGraph is under active development. An alpha version will be released soon with the npm package once [licensing](#license)
+and [versioning](https://github.com/maxGraph/maxGraph/issues/92) issues have been resolved.
+
+In the meantime, you can test the project by running the [Storybook examples](#development).
+
+
+## <a id="license"></a> Transition to Apache 2.0 license
+
+This project is published under the terms of a [Modified-Apache-2.0](LICENSE) license and currently preparing to move back to the standard Apache 2.0 license.  
+From now on, all contributions to this project must be under the terms of the Apaches 2.0 license as stated in https://www.apache.org/licenses/LICENSE-2.0.  
+See https://github.com/maxGraph/maxGraph/issues/89 for more details.
+
+
+## Getting Started
+
+Assuming your page defines an element with the id `graph-container`, the following will display a rectangle connected to an orange circle.
+```typescript
+import {type CellStyle, Graph, InternalEvent} from '@maxgraph/core';
+
+const container = document.getElementById('graph-container') as HTMLElement;
+// Disables the built-in context menu
+InternalEvent.disableContextMenu(container);
+
+// Gets the default parent for inserting new cells. This
+// is normally the first child of the root (ie. layer 0).
+const parent = graph.getDefaultParent();
+graph.setPanning(true); // Use mouse right button for panning
+
+// Adds cells to the model in a single step
+graph.batchUpdate(() => {
+  const vertex01 = graph.insertVertex(parent, null, 'a regular rectangle', 10, 10, 100, 100);
+  const vertex02 = graph.insertVertex(parent, null, 'a regular ellipse', 350, 90, 50, 50, <CellStyle>{shape: 'ellipse', fillColor: 'orange'});
```

Comment:
ℹ️  method signature in the type is `insertVertex: (...args: any[]) => Cell;`
So, cast to `CellStyle` to get code assist.